File: runner.js
class ADKRunner {
    constructor() {
        this.workflow = [];
        this.context = {};
    }

    addAgent(agent) {
        this.workflow.push(agent);
    }

    async execute(initialState) {
        this.context = { ...initialState };
        console.log("🚀 Starting ADK Runner Workflow...\n");
        
        for (const agent of this.workflow) {
            console.log(`🤖 Executing Agent: ${agent.name}`);
            try {
                this.context = await agent.run(this.context);
                console.log(`✅ ${agent.name} Completed.\n`);
            } catch (error) {
                console.error(`❌ Error in ${agent.name}:`, error);
                throw error;
            }
        }
        
        console.log("🎉 Workflow Finished!");
        return this.context;
    }
}

module.exports = ADKRunner;

File: runner.py
class ADKRunner:
    def __init__(self):
        self.workflow = []
        self.context = {}

    def add_agent(self, agent):
        self.workflow.append(agent)

    def execute(self, initial_state):
        self.context = initial_state.copy()
        print("🚀 Starting ADK Runner Workflow...\n")
        
        for agent in self.workflow:
            print(f"🤖 Executing Agent: {agent['name']}")
            try:
                self.context = agent['run'](self.context)
                print(f"✅ {agent['name']} Completed.\n")
            except Exception as e:
                print(f"❌ Error in {agent['name']}: {e}")
                raise e
        
        print("🎉 Workflow Finished!")
        return self.context
